docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/arithmetic/segment-max-16.rst

```diff
@@ -0,0 +1,108 @@
+SegmentMax
+==========
+
+
+.. meta::
+  :description: Learn about SegmentMax-16 - an arithmetic operation which computes the maximum values along segments of a tensor.
+
+**Versioned name**: *SegmentMax-16*
+
+**Category**: *Arithmetic*
+
+**Short description**: *SegmentMax-16* operation finds the maximum value in each specified segment of the ``input`` tensor.
+
+**Detailed description**
+
+For each index in ``segment_ids`` the operator gets values from ``data`` input tensor and calculates the maximum value for each segment.
+
+For example ``segments_ids`` with value ``[0,0,0,1,1,3,5,5]`` defines 4 non-empty segments. When coupled with a 1D data tensor ``data``, the segments are as follows:
+
+* Segment_0: ``[data[0], data[1], data[2]]``
+* Segment_1: ``[data[3], data[4]]``
+* Segment_2: ``[]``
+* Segment_3: ``[data[5]]``
+* Segment_4: ``[]``
+* Segment_5: ``[data[6], data[7]]``
+
+When there are no values in a segment, ``output[segment]`` is defined by ``fill_mode`` attribute.
+
+For ``fill_mode`` equal to ``ZERO`` , the operation output would be ``[max(Segment_0), max(Segment_1), 0, max(Segment_3), 0, max(Segment_5)]``.
+
+**Attributes**:
+
+* **1**: *empty_segment_value*
+
+  * **Description**: The value assigned to segments which are empty. **Required.**
+  * **Range of values**: Name of the mode in string format:
+
+    * ``ZERO`` - the empty segments are filled with zeros.
+    * ``LOWEST`` - the empty segments are filled with the lowest value of the data type *T*.
+  * **Type**: ``string``
```

Comment:
Let it be a constant input (the fourth one) of type *T* please. It will be more flexible for future use cases.
Attribute is injected in IR and no way to change it in run-time

Reply:
Since one of the values is `LOWEST`, which means the lowest value of type `T`, wouldn't this require custom behavior for graph type conversions? Setting that value dynamically during runtime could be safer.

Reply:
I don't know but I don't exclude that this operation can be used in some decompositions to express some new FW operation. If we make an attribute for it, we create additional constraints. I think we don't need addtional constraints and let it make more flexible.